File: password checker.py
import hashlib
import json

#function to hash the password using SHA256
def encodePass(password): 
    passHash = hashlib.sha256(password).hexdigest()
    return passHash

#open a json file with the accounts
with open('accounts.json', 'r') as accounts:
 
    # Reading from json file
    accounts = json.load(accounts)

#visual check to see if persistant data worked
print(accounts)
print(type(accounts))

signIn = True

while signIn:

    choice = input("create an account [1] or sign in as existing user [2]: ")

    if choice == '1':
        username = input("Enter Username: ")
        password = input("Enter Password: ").encode("ascii")
        encryptedPass = encodePass(password)
        
        #add account data to dictonary from json file
        accounts.update({username:encryptedPass})
        print(accounts)
        
        #store account data in json file
        with open('accounts.json', 'w') as convert_file:
            convert_file.write(json.dumps(accounts, indent=4))

    elif choice == '2':
        invalid = True
        while invalid:
            username = input("Enter your username: ")
            password = input("Enter your password: ").encode("ascii")
            encryptedPass = encodePass(password)
            if username in accounts and encryptedPass in accounts.values():
                print("Welcome to DaxSoft")
                invalid = False

            else:
                print("Available accounts: ")
                print(accounts)
                print("Please try again")
    else:
        signIn = False

File: passwordChecker.js

const fs = require('fs');
const crypto = require('crypto');
const readline = require('readline');

function encodePass(password) {
const passHash = crypto.createHash('sha256').update(password).digest('hex');
return passHash;
}

let accounts = JSON.parse(fs.readFileSync('accounts.json', 'utf8'));

async function signInProcess() {

  let invalid = true;
  while (invalid) {
      let username = document.getElementById('username');
      let password = document.getElementById('password');
      let encryptedPass = encodePass(password);
      if (accounts[username] === encryptedPass) {
          invalid = false;
          location.replace("/other.html")
      } else {
          console.log("Please try again");
      }
  }
}
async function createAccount(){
  let username = document.getElementById(username);
  let password = document.getElementById(password);
  let encryptedPass = encodePass(password);

  //add account data to dictionary from json file
  accounts[username] = encryptedPass;
  console.log(accounts);

  //store account data in json file
  fs.writeFileSync('accounts.json', JSON.stringify(accounts, null, 4));
}
async function test(){
location.replace("./other.html")
}
